features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/timeline/factories/virtual/TimelineItemVirtualFactory.kt

```diff
@@ -30,8 +31,13 @@ class TimelineItemVirtualFactory @Inject constructor(
     fun create(
         virtualTimelineItem: MatrixTimelineItem.Virtual,
     ): TimelineItem.Virtual {
+        val id = if (virtualTimelineItem.virtual is VirtualTimelineItem.EncryptedHistoryBanner) {
+            "encrypted_history_banner"
```

Comment:
Thing is, we receive those values from the Rust SDK and we don't really know which ones are available. I tried using `-1` (and I think `-2`?) here and it turned out they were already in use by some virtual events in the Rust SDK, so since component is something that only exists at the app level I just made a special case for it to prevent any future clashes in ids.

Reply:
I see, OK, thanks!